homeassistant/components/sensor/sensehat.py

```diff
@@ -0,0 +1,139 @@
+"""
+Support for Sense HAT sensors.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/
+"""
+import os
+import logging
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['https://github.com/RPi-Distro/python-sense-hat'
+                '/archive/master.zip#sense-hat==2.2.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_cpu_temp():
+    """Get CPU temperature."""
+    res = os.popen("vcgencmd measure_temp").readline()
+    t_cpu = float(res.replace("temp=", "").replace("'C\n", ""))
+    return t_cpu
+
+
+def get_average(temp_base):
```

Comment:
The [`statistics`](https://home-assistant.io/components/sensor.statistics/) sensor would help the users to get the average of a given number of values. Platforms shouldn't implement mathematical operations if not needed.